base/stats.go

```diff
@@ -702,11 +702,47 @@ type SharedBucketImportStats struct {
 	ImportPartitions *SgwIntStat `json:"import_partitions"`
 }
 
+type SgwStatWrapper interface {
+	FormatString() string
+	Name() string
+	Help() string
+	LabelKeys() []string
+	ValueTypeString() string
+}
+
 type SgwStat struct {
-	statFQN       string
-	statDesc      *prometheus.Desc
-	labelValues   []string
-	statValueType prometheus.ValueType
+	fqn       string
+	help      string
+	labels    map[string]string
+	valueType prometheus.ValueType
+	desc      *prometheus.Desc
```

Comment:
Can we rename the changes ones back? Seems unnecessary and adds a load of changes to the PR.